src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -936,13 +916,21 @@ void C2_MacroAssembler::fast_unlock(Register objReg, Register boxReg, Register t
   testl (boxReg, 0);                      // set ICC.ZF=1 to indicate success
   jmpb  (DONE_LABEL);
 
+#endif
   if (!UseHeavyMonitors) {
     bind  (Stacked);
     movptr(tmpReg, Address (boxReg, 0));      // re-fetch
     lock();
     cmpxchgptr(tmpReg, Address(objReg, oopDesc::mark_offset_in_bytes())); // Uses RAX which is box
+    // Intentional fall-thru into DONE_LABEL
   }
-#endif
+
+  // TODO: Comment still valid?
```

Comment:
We have the same comment here:
https://github.com/openjdk/jdk/blob/12e76cbc725ff87577e2ef23267590eae37a82d1/src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp#L707

And it seems to be there from the beginning:
https://hg.openjdk.java.net/jdk/jdk/annotate/74aaad871363/hotspot/src/cpu/x86/vm/x86_32.ad#l3441

I think instead of adding a `TODO`, we should either completely remove the comments or file a follow-up enhancement to investigate.

Reply:
I decided to remove the TODO. If we were to optimize this, then there are probably lower-hanging fruits like looking at the subsequent branch after the DONE_LABEL where we jump around the counting code.